lib/Remotery.c

```diff
@@ -4233,11 +4233,11 @@ static void Remotery_Destructor(Remotery* rmt)
     // Join the remotery thread before clearing the global object as the thread is profiling itself
     Delete(Thread, rmt->thread);
 
-    // Ensure this is the module that created it
-    assert(g_RemoteryCreated == RMT_TRUE);
-    assert(g_Remotery == rmt);
-    g_Remotery = NULL;
-    g_RemoteryCreated = RMT_FALSE;
+    if (g_RemoteryCreated)
+    {
+      g_Remotery = NULL;
+      g_RemoteryCreated = RMT_FALSE;
+    }
 
```

Comment:
Yes that makes sense.

I wrote it in this way to mimick the New() .. Delete(...) pattern which is used for other aspects of the remotery instance to deal with conditional destruction of a constructed part.

I.e. the pattern that Delete(...) follows of destructing iff the object is not null